src/modules/roc_netio/target_uv/roc_netio/handle.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_netio/target_uv/roc_netio/handle.h
+//! @brief Handle.
+
+#ifndef ROC_NETIO_HANDLE_H_
+#define ROC_NETIO_HANDLE_H_
+
+#include "roc_packet/address.h"
+
+namespace roc {
+namespace netio {
+
+//! Network handle.
+struct Handle {
```

Comment:
The name is confusing. A "handle" is usually an (opaque) reference to some resource, e.g. file handle or window handle.. This struct is something more like a callback/delegate or notification handleR.

But I suggest to get rid of this struct at all and replace it with an interface like:

```
class ICloseHandler {
public:
    virtual void closed(...) = 0;
}
```

Then Transceiver can (privately!) implement this interface and pass itself to UDPReceiver and UDPSender.

This approach is (IMHO) a bit cleaner and also easier to analyze because you can quickly check what classes implement ICloseHandler.

Reply:
done